### ISSUE TITLE
Modal improvements

### ISSUE THREAD

**Issue by veekeys** [iTwin/iTwinUI#608]
Do a full review on the code, sturcture, specificity and try to improve it without delivering a breaking change to react.

**Comment by FlyersPh9** [comment]
- Add `.iui-modal-content` to all modal types.  Currently default does not have it.
- Add the option for an icon in front of the title.  #586 
- Add the option for more than 1 icon in the top right corner.
![Screen Shot 2022-04-29 at 3 32 28 PM](https://user-images.githubusercontent.com/849817/166057544-32ff297c-3798-4f03-b828-7fd076e473c0.png)

**Comment by mayank99** [comment]
I think this issue is no longer relevant. `ModalContent` is usable everywhere, and text-with-icon will be tracked in #586, and multiple icon support is possible after #729.
